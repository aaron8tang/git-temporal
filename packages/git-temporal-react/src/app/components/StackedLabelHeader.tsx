import React from 'react';

import { StackedLabel, StackedLabelProps } from './StackedLabel';
import { style } from 'app/styles';

const localStyles = {
  labelStyle: {
    _extends: 'h3Text',
    marginBottom: 5,
  },
  textStyle: {
    _extends: 'h2Text',
  },
};

export const StackedLabelHeader = (props: StackedLabelProps): JSX.Element => {
  const propsForward = Object.assign({}, props, {
    labelStyle: [localStyles.labelStyle, props.labelStyle],
  });

  return (
    <StackedLabel {...propsForward}>
      <div style={style(localStyles.textStyle)}>{props.children}</div>
    </StackedLabel>
  );
};
