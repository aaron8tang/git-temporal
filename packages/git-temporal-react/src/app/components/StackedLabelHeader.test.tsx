import React from 'react';
import { shallow } from 'enzyme';

import { StackedLabelHeader } from './StackedLabelHeader';

describe('components/StackedLabelHeader', () => {
  describe('when rendered with required props', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <StackedLabelHeader label="This is the Label">
          this is the text
        </StackedLabelHeader>
      );
    });

    test('it should match snapshot (it should have rendered a label and a text element)', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
