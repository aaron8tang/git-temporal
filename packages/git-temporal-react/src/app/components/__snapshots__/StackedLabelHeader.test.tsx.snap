// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/StackedLabelHeader when rendered with required props it should match snapshot (it should have rendered a label and a text element) 1`] = `
<StackedLabel
  label="This is the Label"
  labelStyle={
    Object {
      "fontSize": 14,
      "fontWeight": "bold",
      "lineHeight": 1.35,
      "marginBottom": 5,
      "marginRight": "15px",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": 18,
        "fontWeight": "bold",
        "lineHeight": 1.35,
        "marginBottom": "15px",
        "marginRight": "15px",
      }
    }
  >
    this is the text
  </div>
</StackedLabel>
`;
